packages/starknet-snap/src/ui/utils.tsx

```diff
@@ -80,3 +92,149 @@ export async function generateExecuteTxnFlow(
     },
   });
 }
+
+/**
+ * Update the interface for a ExecuteTxnUI
+ *
+ * @param id - Interface Id
+ * @param request - TransactionRequest
+ * @param errors
+ * @param errors.errors
+ */
+export async function updateExecuteTxnFlow(
+  id: string, // Interface Id to update
+  request: TransactionRequest, // Props must include `id` and `interfaceId`
+  errors?: { errors: ExecuteTxnUIErrors }, // Optional partial props for error handling or overrides
+) {
+  const {
+    signer,
+    chainId,
+    networkName,
+    maxFee,
+    calls,
+    selectedFeeToken,
+    includeDeploy,
+  } = request;
+  // Perform the interface update
+  await snap.request({
+    method: 'snap_updateInterface',
+    params: {
+      id,
+      ui: (
+        <ExecuteTxnUI
+          signer={signer}
+          chainId={chainId}
+          networkName={networkName}
+          maxFee={maxFee}
+          calls={calls}
+          selectedFeeToken={selectedFeeToken}
+          includeDeploy={includeDeploy}
+          {...errors}
+        />
+      ),
+    },
+  });
+}
+
+export type UpdateInterfaceParams = {
+  id: string;
+  ui: JSX.Element;
+};
+
+/**
+ *
+ * @param id
+ * @param ui
+ */
+export async function updateInterface(
+  id: string,
+  ui: JSX.Element,
+): Promise<void> {
+  await snap.request({
+    method: 'snap_updateInterface',
+    params: {
+      id,
+      ui,
+    },
+  });
+}
+
+/**
+ * Checks if the provided address has sufficient funds in either STRK or ETH to cover
+ * the total amount required for a set of contract calls and a suggested transaction fee.
+ *
+ * @param address - The wallet address to check balances for.
+ * @param network - The network object, which includes information such as chainId.
+ * @param calls - An array of contract call objects.
+ * Each object should contain a contract address and an optional amount.
+ * @param feeToken - Specifies the fee token type (either STRK or ETH) to use for the transaction fee.
+ * @param suggestedMaxFee - The maximum transaction fee to consider, as a stringified bigint.
+ * @returns Resolves to `true` if there are sufficient funds for the specified fee token
+ * to cover both the total amounts in `calls` and the `suggestedMaxFee`; otherwise, `false`.
+ */
+export async function hasSufficientFunds(
+  address: string,
+  network: Network,
+  calls: { contractAddress: string; amount?: string }[],
+  feeToken: FeeToken,
+  suggestedMaxFee: string,
+): Promise<boolean> {
+  const stateManager = new TokenStateManager();
+
+  // Fetch STRK token details and balance
+  const strkToken = await stateManager.getStrkToken({
+    chainId: network.chainId,
+  });
+  const strkBalance = strkToken
```

Comment:
nit:

if we have the token and network information in the context, then we dont need to fetch it here
i remember why you wanna to format the call, because we dont wanna do it over and over